controllers/machine_controller.go

```diff
@@ -339,32 +350,26 @@ func (r *MachineReconciler) drainNode(cluster *clusterv1.Cluster, nodeName strin
 
 func (r *MachineReconciler) deleteNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
 	logger := r.Log.WithValues("machine", name)
-
+	var c client.Client
```

Comment:
```suggestion
	c := r.Client
```

and then change the `if` condition to overwrite this value if the cluster isn't nil, removing the need of an else